cmd/eval-dev-quality/cmd/evaluate.go

```diff
@@ -502,6 +513,108 @@ func (command *Evaluate) evaluateDocker(ctx *evaluate.Context) (err error) {
 	return nil
 }
 
+// evaluateKubernetes executes the evaluation for each model inside a kubernetes run container.
+func (command *Evaluate) evaluateKubernetes(ctx *evaluate.Context) (err error) {
+	tmpl, err := template.ParseFiles(filepath.Join("conf", "kubernetes-job.yml"))
+	if err != nil {
+		return pkgerrors.Wrap(err, "could not create kubernetes job template")
+	}
+
+	// Filter all the args to pass them onto the container.
+	args := util.FilterArgs(os.Args[2:], []string{
+		"--model",
+		"--parallel",
+		"--result-path",
+		"--runtime",
+	})
+
+	parallel := util.NewParallel(command.Parallel)
+
+	// Iterate over each model and start the container.
+	for i, model := range ctx.Models {
+		// We are skipping ollama models until we fully support pulling. https://github.com/symflower/eval-dev-quality/issues/100.
+		if ctx.ProviderForModel[model].ID() == "ollama" {
+			command.logger.Print("Skipping unsupported ollama model with docker runtime")
+
+			continue
+		}
+
+		// Commands regarding the docker runtime.
+		kubeCommand := []string{
+			"kubectl",
+			"apply",
+			"-f",
+			"-", // apply STDIN
+		}
+
+		// Commands for the evaluation to run inside the container.
+		evaluationCommand := []string{
+			"eval-dev-quality",
+			"evaluate",
+			"--model",
+			model.ID(),
+			"--result-path",
+			"/var/evaluations/" + model.ID(),
+		}
+		cmd := append(evaluationCommand, args...)
+
+		// Template data
+		nameReplacer := strings.NewReplacer(
+			"/", "-",
+			"\\", "-",
+			":", "-",
+		)
+		jobName := fmt.Sprintf("%s-%v", nameReplacer.Replace(model.ID()), i)
+		data := map[string]any{
+			"name":      jobName,
+			"namespace": command.Namespace,
+			"image":     command.RuntimeImage,
+			"command":   `["` + strings.Join(cmd, `","`) + `"]`,
+		}
+
+		parallel.Execute(func() {
+			var tmplData bytes.Buffer
+			tmpl.Execute(&tmplData, data)
+
+			commandOutput, err := util.CommandWithResult(context.Background(), command.logger, &util.Command{
+				Command: kubeCommand,
+				Stdin:   tmplData.String(),
+			})
+			if err != nil {
+				err = pkgerrors.WithMessage(pkgerrors.WithStack(err), commandOutput)
+				command.logger.Printf("ERROR: %s", err)
```

Comment:
should return here and not continue

same is probably true for the local `docker`  runtime... can you check?